### ISSUE TITLE
Print warning when run against an unsupported file

### ISSUE THREAD

**Issue by not-my-profile** [charliermarsh/ruff#1473]
```
$ echo 'not json' > foo.json 
$ ruff foo.json 
$ echo $?
0
```

We might also consider returning a non-zero status code since the file isn't supported.

**Comment by charliermarsh** [comment]
Good idea.

**Comment by charliermarsh** [comment]
(And agree -- non-zero.)

**Comment by charliermarsh** [comment]
We should probably also warn if we can't find any valid files (but aren't passed an invalid file _directly_).

For example: `ruff foo/` where `foo` is an empty directory, or doesn't contain any `.py` or `.pyi` files.

Here's how Black handles these cases:

```
ruff on  charlie/fix-message:main [$⇡] is 📦 v0.0.202 via 🐍 v3.11.1 (.venv) via 🦀 v1.65.0
❯ black pyproject.toml
error: cannot format pyproject.toml: Cannot parse: 41:8: combine-as-imports = true

Oh no! 💥 💔 💥
1 file failed to reformat.
ruff on  charlie/fix-message:main [$⇡] is 📦 v0.0.202 via 🐍 v3.11.1 (.venv) via 🦀 v1.65.0
❯ black pyproject.toml setup.py
blerror: cannot format pyproject.toml: Cannot parse: 41:8: combine-as-imports = true

Oh no! 💥 💔 💥
1 file left unchanged, 1 file failed to reformat.
ruff on  charlie/fix-message:main [$⇡] is 📦 v0.0.202 via 🐍 v3.11.1 (.venv) via 🦀 v1.65.0
❯ black foo
No Python files are present to be formatted. Nothing to do 😴
ruff on  charlie/fix-message:main [$⇡] is 📦 v0.0.202 via 🐍 v3.11.1 (.venv) via 🦀 v1.65.0
❯ black foo/ setup.py
All done! ✨ 🍰 ✨
1 file left unchanged.
```

**Comment by charliermarsh** [comment]
In short:

- (Optional) If we don't find _any_ files, show a warning.
- If we're passed a non `.py` or `.pyi` file, raise an error for that file (but process any valid files).
